src/benchmarklib/table_builder.hpp

```diff
@@ -0,0 +1,241 @@
+    return {};
+  }
+};
+
+template <typename T, typename Enable = void>
+struct is_optional : std::false_type {};
+
+template <typename T>
+struct is_optional<T, std::enable_if_t<std::is_same_v<T, std::optional<typename T::value_type>>>> : std::true_type {};
+
+// is_optional_v<T> <=> T is of type std::optional<?>
+template <typename T>
+constexpr bool is_optional_v = is_optional<T>::value;
+
+template <typename T, typename Enable = void>
+struct GetValueType {
+  using value_type = T;
+};
+
+template <typename T>
+struct GetValueType<T, std::enable_if_t<is_optional_v<T>>> {
+  using value_type = typename T::value_type;
+};
+
+// get the value_type (lets call it V) of std::optional<V> and V using the same syntax - needed for templating
+template <typename T>
+using get_value_type = typename GetValueType<T>::value_type;
+
+// check if std::optional<V> or V is null using the same syntax - needed for templating
+template <typename T>
+constexpr bool is_null(const T& maybe_optional) {
+  if constexpr (is_optional_v<T>) {
+    return !maybe_optional.has_value();
+  } else {
+    return false;
+  }
+}
+
+// get the value of std::optional<V> or V using the same syntax - needed for templating
+template <typename T>
+get_value_type<T>& get_value(T& maybe_optional) {
+  if constexpr (is_optional_v<T>) {
+    return maybe_optional.value();
+  } else {
+    return maybe_optional;
+  }
+}
+
+}  // namespace table_builder
+
+/**
+ * Helper to build a table with a static column layout, specified by constructor args types and names. Keeps a
+ * value vector for each column and appends values to them in append_row(). For nullable columns an additional
+ * null_values vector is kept. Automatically creates chunks in accordance with the specified chunk size.
+ */
+template <typename... DataTypes>
+class TableBuilder {
+ public:
+  // names is a boost::hana::tuple of strings defining column names, types is a list of equal length defining respective
+  // column types types may contain std::optional<?>, which will result in a nullable column, otherwise columns are not
+  // nullable
+  template <typename Names>
+  TableBuilder(size_t chunk_size, const boost::hana::tuple<DataTypes...>& types, const Names& names,
+               opossum::UseMvcc use_mvcc, size_t estimated_rows = 0)
+      : _use_mvcc(use_mvcc), _estimated_rows_per_chunk(std::min(estimated_rows, chunk_size)) {
+    BOOST_HANA_CONSTANT_ASSERT(boost::hana::size(names) == boost::hana::size(types));
+
+    // Iterate over the column types/names and create the columns.
+    auto column_definitions = TableColumnDefinitions{};
+    boost::hana::for_each(boost::hana::zip(names, types), [&](const auto& name_and_type) {
+      auto name = name_and_type[boost::hana::llong_c<0>];
+      auto type = name_and_type[boost::hana::llong_c<1>];
+
+      auto data_type = data_type_from_type<table_builder::get_value_type<decltype(type)>>();
+      auto is_nullable = table_builder::is_optional_v<decltype(type)>;
```

Comment:
if you need is_optional_v just here, maybe put its decl here to keep things compact?

Reply:
It is needed elsewhere too.